src/gpu/intel/jit/gemm/jit_gemm_pd.cpp

```diff
@@ -160,6 +160,181 @@ status_t jit_gemm_pd_t::init_post_ops() {
+                if (zp_group_k_a_ < d->k()) {
+                    wei_zp_2d_ = true;
+                    ao_dims_ = 2;
+                    wei_q2d_group_k_ = zp_group_k_a_;
+                }
+            }
+        }
+        if (!attr_zps.has_default_values(DNNL_ARG_B)) {
+            cmask_b_ = attr_zps.get_mask(DNNL_ARG_B);
+            bo_dims_ = cmask_b_ > 0;
+            if (!attr_zps.has_default_groups(DNNL_ARG_B)) {
+                zp_group_k_b_ = attr_zps.get_group(DNNL_ARG_SRC, 1);
+                if (zp_group_k_b_ < d->k()) {
+                    bo_dims_ = 2;
+                    src_q2d_group_k_ = zp_group_k_b_;
+                }
+            }
+        }
+        if (!attr_zps.has_default_values(DNNL_ARG_C))
+            cmask_c_ = attr_zps.get_mask(DNNL_ARG_C);
+    }
+
+    const auto *wei_scales = &attr()->scales_.get(DNNL_ARG_WEIGHTS);
+    const auto *src_scales = &attr()->scales_.get(DNNL_ARG_SRC);
+
+    if (!wei_scales->has_default_values()) {
+        wei_scales_type_ = wei_scales->get_data_type();
+        if (!wei_scales->has_default_groups()) {
+            wei_scales_group_k_ = wei_scales->get_group(0);
+            if (quant_enabled_ && !wei_scales->has_default_groups()
+                    && wei_scales_group_k_ < d->k()) {
+                wei_scales_2d_ = true;
+                if (!wei_zp_2d_) wei_q2d_group_k_ = wei_scales_group_k_;
+            }
+        }
+    }
+
+    if (!src_scales->has_default_values()) {
+        src_scales_type_ = src_scales->get_data_type();
+        src_po_sc_ = src_scales->get_mask() == 2;
+        if (!src_scales->has_default_groups()) {
+            src_scales_group_k_ = src_scales->get_group(1);
+            if (quant_enabled_ && !src_scales->has_default_groups()
+                    && src_scales_group_k_ < d->k()) {
+                src_scales_2d_ = true;
+                src_q2d_group_k_ = src_scales_group_k_;
+            }
+        }
+    }
+}
+
+bool jit_gemm_pd_t::zp_ok() {
+    auto &attr_zps = attr()->zero_points_;
+    int ndims = desc()->a_desc.ndims;
+    const auto d = desc();
+    using namespace data_type;
+
+    if (!attr_zps.has_default_values(DNNL_ARG_A)) {
+        // Groups determine supported masks.
+        if (!attr_zps.has_default_groups(DNNL_ARG_A)) {
+            if (!valid_2d_mask(cmask_a_, ndims)) return false;
+            const auto wei_q2d_group_n
+                    = attr_zps.get_group(DNNL_ARG_WEIGHTS, 1);
+            // Non-trivial N group unsupported.
+            if (wei_q2d_group_n != 1) return false;
+            // Zero points with non-trivial groups only supported
+            // when target tensor is being dequantized.
+            if (!(!dy_quant_enabled_ || utils::one_of(d->a_type(), s4, u4)
+                        || zp_group_k_a_ == desc()->k()))
+                return false;
+        } else {
+            if (!utils::one_of(cmask_a_, 0, mask_per_oc, mask_per_ic))
+                return false;
```

Comment:
I'm pretty sure, based on how `cmask_a_` is constructed, it will always be `-1` in this block. Since it's default-constructed to `-1`, and only updated if `!attr_zps.has_default_values(DNNL_ARG_A)`.

Reply:
It's very possible, I misjudged the indentation level :)